crates/autopilot/src/database/quotes.rs

```diff
@@ -16,6 +19,29 @@ impl Postgres {
         database::quotes::remove_expired_quotes(&mut ex, max_expiry).await?;
         Ok(())
     }
+
+    pub async fn read_quotes(&self, auction: &Auction) -> Result<HashMap<OrderUid, Quote>> {
+        let mut quote_tasks = Vec::new();
+        for order in &auction.orders {
+            let mut ex = self.pool.acquire().await?;
+            let order_uid = ByteArray(order.metadata.uid.0);
+            let quote_task = tokio::spawn(async move {
+                database::orders::read_quote(&mut ex, &order_uid)
+                    .await
+                    .ok()?
+            });
```

Comment:
You usually only want to use `tokio::spawn()` if the runtime is supposed to start working on the future in the background without having to `.await` it.
In this case it makes more sense to build an interator that yields futures. Then you can await all the futures in parallel using `futures::future::join_all()`.
FYI it's possible to write the future iterator in a way that allows collecting into a hashmap.

Reply:
Will try with sequential call first and evaluate the performance